test/functional/ex_cmds/cd_spec.lua

```diff
@@ -0,0 +1,101 @@
+-- Specs for :cd, :tcd, :lcd and hastabdir(), haslocaldir()
+
+local helpers = require('test.functional.helpers')
+local execute, eq, clear, eval, feed =
+  helpers.execute, helpers.eq, helpers.clear, helpers.eval, helpers.feed
+
+
+-- These directories will be created for testing
+local directories = {
+  'Xtest-functional-ex_cmds-cd_spec.1', -- Tab
+  'Xtest-functional-ex_cmds-cd_spec.2', -- Window
+  'Xtest-functional-ex_cmds-cd_spec.3', -- New global
+}
+
+-- Shorthand writing to get the current working directory
+local  cwd = function() return eval('getcwd()'   ) end  -- working dir
+local gcwd = function() return eval('getcwd("g")') end  -- global dir
+local tcwd = function() return eval('getcwd("t")') end  -- tab dir
+local wcwd = function() return eval('getcwd("w")') end  -- window dir
+
+-- Test both the `cd` and `chdir` variants
+for _, cmd in ipairs {'cd', 'chdir'} do
+  describe(':*' .. cmd, function()
+    before_each(function()
+      clear()
+      for _, d in ipairs(directories) do
+        lfs.mkdir(d)
+      end
+    end)
+
+    after_each(function()
+      for _, d in ipairs(directories) do
+        lfs.rmdir(d)
+      end
+    end)
+
+    it('works', function()
+      -- Store the initial working directory
+      local globalDir = cwd()
+
+      -- Create a new tab first and verify that is has the same working dir
+      execute('tabnew')
+      eq(globalDir, cwd())
+      eq('', tcwd())  -- has neither tab- nor window-local directory
+      eq('', wcwd())
+
+      -- Change tab-local working directory and verify it is different
+      execute('t' .. cmd .. ' ' .. directories[1])
+      eq(globalDir .. '/' .. directories[1], cwd())
+      eq(cwd(), tcwd())  -- working directory maches tab directory
+      eq(   '', wcwd())  -- still no window-directory
```

Comment:
`eq(''`, not `eq(<spaces>''`. Latter is not readable, though not as unreadable as `eq(                             cwd(),`.